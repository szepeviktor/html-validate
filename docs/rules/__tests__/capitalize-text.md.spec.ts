import HtmlValidate from "../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["incorrect"] = `<h1>lorem ipsum</h1>`;
markup["correct"] = `<h1>Lorem ipsum</h1>`;
markup["ignored"] = `<h1><code>loremIpsum</code> dolor sit amet</h1>`;

describe("docs/rules/capitalize-text.md", () => {
	it("inline validation: incorrect", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"capitalize-text":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"capitalize-text":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: ignored", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"capitalize-text":"error"}});
		const report = htmlvalidate.validateString(markup["ignored"]);
		expect(report.results).toMatchSnapshot();
	});
});
