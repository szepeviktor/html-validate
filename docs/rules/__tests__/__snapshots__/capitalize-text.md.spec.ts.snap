// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs/rules/capitalize-text.md inline validation: correct 1`] = `Array []`;

exports[`docs/rules/capitalize-text.md inline validation: ignored 1`] = `Array []`;

exports[`docs/rules/capitalize-text.md inline validation: incorrect 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "inline",
    "messages": Array [
      Object {
        "column": 2,
        "context": Object {
          "tagName": "h1",
        },
        "line": 1,
        "message": "Text must be capitalized in <h1>",
        "offset": 1,
        "ruleId": "capitalize-text",
        "ruleUrl": "https://html-validate.org/rules/capitalize-text.html",
        "selector": "h1",
        "severity": 2,
        "size": 2,
      },
    ],
    "source": "<h1>lorem ipsum</h1>",
    "warningCount": 0,
  },
]
`;
